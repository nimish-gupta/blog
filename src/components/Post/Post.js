// @flow strict
import React from 'react';
import Author from './Author';
import Comments from './Comments';
import Content from './Content';
import Meta from './Meta';
import Tags from './Tags';
import * as Styled from './Post.style';
import type { Node } from '../../types';

type Props = {
	post: Node,
};

const Post = ({ post }: Props) => {
	const { html } = post;
	const { tagSlugs, slug } = post.fields;
	const { tags, title, date } = post.frontmatter;

	return (
		<div>
			<Styled.HomeButton to="/">All Articles</Styled.HomeButton>

			<div>
				<Content body={html} title={title} />
			</div>

			<Styled.Footer>
				<Meta date={date} />
				{tags && tagSlugs && <Tags tags={tags} tagSlugs={tagSlugs} />}
				<Author />
			</Styled.Footer>

			<Styled.Comment>
				<Comments postSlug={slug} postTitle={post.frontmatter.title} />
			</Styled.Comment>
		</div>
	);
};

export default Post;
