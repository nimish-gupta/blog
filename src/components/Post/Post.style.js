import styled from 'styled-components';
import { Link } from 'gatsby';

import { breakpointMd, marginTop } from '../../assets/styled-components/mixins';
import Toggle from '../Sidebar/Toggle';

export const Footer = styled.div`
	max-width: ${(props) => props.theme.layout.postWidth};
	margin: 0 auto;
	padding: 0 15px;
	${breakpointMd`
		padding: 0;
	`}
`;

export const Comment = styled.div`
	max-width: ${(props) => props.theme.layout.postWidth};
	margin: 0 auto;
	padding: 0 15px;
	${breakpointMd`
		padding: 0;
	`}
`;

export const HomeButton = styled(Link)`
	display: block;
	max-width: 90px;
	height: ${(props) => props.theme.button.height};
	padding: 0 24px;
	line-height: ${(props) => props.theme.button.height};
	text-align: center;
	color: ${(props) => props.theme.color.base};
	border: 1px solid ${(props) => props.theme.color.grayBorder};
	border-radius: ${(props) => props.theme.button.borderRadius};
	font-size: ${(props) => props.theme.typographic.baseFontSize};
	font-weight: normal;
	margin-left: auto;
	margin-right: auto;
	${marginTop(1)};

	&:hover,
	&:focus {
		color: ${(props) => props.theme.color.primary};
	}
	${breakpointMd`
		position: fixed;
      max-width: auto;
      margin: 0;
      top: 30px;
      left: 30px
	`}
`;

export const Switch = styled(Toggle)`
	margin-left: auto;
	margin-right: auto;
	${breakpointMd`
		position: fixed;
		margin: 0;
		top: 82px;
		left: 30px;
	`};
`;
