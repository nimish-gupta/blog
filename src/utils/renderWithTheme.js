import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import theme from '../assets/styled-components/_variables';

export default function renderWithTheme(component) {
	return renderer.create(
		<ThemeProvider theme={theme}>{component}</ThemeProvider>
	);
}
