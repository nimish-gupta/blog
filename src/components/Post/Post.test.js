// @flow strict
import React from 'react';
import { useStaticQuery, StaticQuery } from 'gatsby';
import renderWithTheme from '../../utils/renderWithTheme';
import Post from './Post';
import siteMetadata from '../../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../../types';

describe('Post', () => {
	beforeEach(() => {
		StaticQuery.mockImplementationOnce(
			({ render }: RenderCallback) => render(siteMetadata),
			useStaticQuery.mockReturnValue(siteMetadata)
		);
	});

	const props = {
		post: {
			id: 'test-123',
			html: '<p>test</p>',
			fields: {
				slug: '/test',
				categorySlug: '/test-category',
				tagSlugs: ['/test_0', '/test_1'],
				readingTime: {
					text: '1 min read',
				},
			},
			frontmatter: {
				date: '2016-09-01',
				tags: ['test_0', 'test_1'],
				title: 'test',
			},
		},
	};

	it('renders correctly', () => {
		const tree = renderWithTheme(<Post {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
