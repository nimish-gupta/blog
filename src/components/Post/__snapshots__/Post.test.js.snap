// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders correctly 1`] = `
<div>
  <a
    className="sc-fzplWN bDGJkX"
    href="/"
  >
    All Articles
  </a>
  <div>
    <div
      className="sc-AxiKw gcwQvO"
    >
      <h1
        className="sc-AxhCb dwCsaS"
      >
        test
      </h1>
      <div
        className="sc-AxhUy eoQtdv"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>test</p>",
          }
        }
      />
    </div>
  </div>
  <div
    className="sc-fzoLsD beBCOR"
  >
    <div>
      <p
        className="sc-AxgMl hlmNXO"
      >
        Published 
        1 Sep 2016
      </p>
    </div>
    <div
      className="sc-AxheI icmSOO"
    >
      <ul
        className="sc-Axmtr fxZEnh"
      >
        <li
          className="sc-AxmLO hoepre"
        >
          <a
            className="sc-fzozJi hJwiIW"
            href="/test_0"
          >
            test_0
          </a>
        </li>
        <li
          className="sc-AxmLO hoepre"
        >
          <a
            className="sc-fzozJi hJwiIW"
            href="/test_1"
          >
            test_1
          </a>
        </li>
      </ul>
    </div>
    <div
      className="sc-AxjAm fPmySw"
    >
      <p>
        Test bio
        <a
          className="sc-AxirZ fSOCfz"
          href="https://www.twitter.com/#"
          rel="noopener noreferrer"
          target="_blank"
        >
          <strong>
            Test name
          </strong>
           on Twitter
        </a>
      </p>
    </div>
  </div>
  <div
    className="sc-fzpans izjeTV"
  />
</div>
`;
